pricenode/src/main/java/bisq/price/spot/providers/Poloniex.java

```diff
@@ -19,76 +19,28 @@
 
 import bisq.price.spot.ExchangeRate;
 import bisq.price.spot.ExchangeRateProvider;
-import bisq.price.util.Altcoins;
 
-import org.knowm.xchange.currency.Currency;
-import org.knowm.xchange.currency.CurrencyPair;
-import org.knowm.xchange.poloniex.dto.marketdata.PoloniexMarketData;
-import org.knowm.xchange.poloniex.dto.marketdata.PoloniexTicker;
+import org.knowm.xchange.poloniex.PoloniexExchange;
 
-import org.springframework.core.ParameterizedTypeReference;
 import org.springframework.core.annotation.Order;
-import org.springframework.http.RequestEntity;
 import org.springframework.stereotype.Component;
-import org.springframework.web.client.RestTemplate;
-import org.springframework.web.util.UriComponentsBuilder;
 
 import java.time.Duration;
 
-import java.util.Date;
-import java.util.Map;
 import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 @Component
 @Order(4)
-class Poloniex extends ExchangeRateProvider {
-
-    private final RestTemplate restTemplate = new RestTemplate();
+public class Poloniex extends ExchangeRateProvider {
```

Comment:
Prefer package-private visibility whenever possible, i.e. please remove the `public` modifier here and in all other `ExchangeRateProvider` implementations. The `bisq.spot.providers` package is expressly designed such that all member classes are package-private; they need not and should not be referenced from any other package.